src/kv.rs

```diff
@@ -72,14 +71,12 @@ pub enum Operation {
 // Helper to extract key value operation from message headers
 fn kv_operation_from_maybe_headers(maybe_headers: Option<&HeaderMap>) -> Operation {
     if let Some(headers) = maybe_headers {
-        if let Some(set) = headers.get(KV_OPERATION) {
-            if set.get(KV_OPERATION_DELETE).is_some() {
-                return Operation::Delete;
-            }
-
-            if set.get(KV_OPERATION_PURGE).is_some() {
-                return Operation::Purge;
-            }
+        if let Some(op) = headers.get(KV_OPERATION) {
+            return match op.as_str() {
+                KV_OPERATION_DELETE => Operation::Delete,
+                KV_OPERATION_PURGE => Operation::Purge,
+                _ => Operation::Put,
```

Comment:
is this something we missed in ADR, that default behaviour is PUT if no operation header is set?

Reply:
Behavior is the same as we've shipped: https://github.com/nats-io/nats.rs/blob/main/src/kv.rs#L86 just rewritten to a single return path with match.

Go defaults to Put https://github.com/nats-io/nats.go/blob/b16cefd916ef7f13d3858627cbcf2538108bd445/kv.go#L173